src/routing_client.rs

```diff
@@ -0,0 +1,210 @@
+
+extern crate crust;
+extern crate sodiumoxide;
+extern crate cbor;
+extern crate maidsafe_types;
+
+use sodiumoxide::crypto;
+use maidsafe_types::Random;
+use std::sync::mpsc::{Receiver, Sender};
+use std::sync::mpsc;
+use std::net::{TcpStream};
+use types;
+use Facade;
+use message_header;
+use messages;
+use maidsafe_types::traits::RoutingTrait;
+
+/// DHT node
+pub struct RoutingClient<'a> {
+    facade: &'a (Facade + 'a),
+    sign_public_key: crypto::sign::PublicKey,
+    sign_secret_key: crypto::sign::SecretKey,
+    encrypt_public_key: crypto::asymmetricbox::PublicKey,
+    encrypt_secret_key: crypto::asymmetricbox::SecretKey,
+    sender: Sender<TcpStream>,
+    receiver: Receiver<TcpStream>,
+    //connection_manager: crust::connection_manager::ConnectionManager,
+    own_address: Vec<u8>,
+    message_id: u32,
+}
+
+impl<'a> RoutingClient<'a> {
+    pub fn new(my_facade: &'a Facade) -> RoutingClient<'a> {
+      sodiumoxide::init(); // enable shared global (i.e. safe to mutlithread now)
+      let key_pair = crypto::sign::gen_keypair();
+      let encrypt_key_pair = crypto::asymmetricbox::gen_keypair();
+      let (tx, rx) : (Sender<TcpStream>, Receiver<TcpStream>) = mpsc::channel();
+
+      RoutingClient { facade: my_facade,
+                    sign_public_key: key_pair.0, sign_secret_key: key_pair.1,
+                    encrypt_public_key: encrypt_key_pair.0, encrypt_secret_key: encrypt_key_pair.1, sender: tx, receiver: rx,
+                    own_address: types::generate_random_vec_u8(64),
+                    message_id: 0,
+      }
+    }
+
+    /// Retreive something from the network (non mutating) - Direct call
+    pub fn get(&mut self, type_id: u64, name: types::DhtId) {
+        // Make GetData message
+        let get_data = messages::get_data::GetData {
+            requester: types::SourceAddress {
+                from_node: self.own_address.clone(), // Should be boost-strap node address ? - will be given the bootstrap node
+                from_group: vec![0; 64], // Dont now what is this so making it invalid // - Option::None
+                reply_to: self.own_address.clone(),
+            },
+            name_and_type_id: types::NameAndTypeId {
+                name: name.0.clone(),
+                type_id: type_id as u32,
+            },
+        };
+
+        // Serialise GetData message
+        let mut encoder = cbor::Encoder::from_memory();
+        encoder.encode(&[&get_data]).unwrap();
+
+        // Make MessageHeader
+        let header = message_header::MessageHeader::new(
+            self.message_id,
+            types::DestinationAddress {
+                dest: name.0,
+                reply_to: vec![0; 64], // None // - Option::None
+            },
+            get_data.requester.clone(),
+            types::Authority::Client,
+            types::Signature::generate_random(), // What to do here?  // Shouldn't be present in MessageHeader
```

Comment:
signature is not required for get Messages. MessageHeader should have signature as optional field 
https://github.com/chandraprakash/MaidSafe-Routing/blob/inetic-MAID-843-integration-with-crux/src/maidsafe/routing/message_header.h#L173